### PR TITLE
support service account name

### PR DESCRIPTION
**Pull request by yanghui16355**
support passing service account name to operator definition

### CODE REVIEW

File: api/v1beta1/flinkcluster_types.go [hunk 1]
```diff
@@ -411,6 +411,10 @@ type FlinkClusterSpec struct {
 	// Flink image spec for the cluster's components.
 	Image ImageSpec `json:"image"`
 
+	// Specifies serviceAccount name assigned to pods
```
Comment (functicons) [inline api/v1beta1/flinkcluster_types.go:414]: "The service account assigned to JobManager, TaskManager and Job submitter Pods. If empty, the default service account in the namespace will be used".

File: api/v1beta1/flinkcluster_types.go [hunk 2]
```diff
@@ -411,6 +411,10 @@ type FlinkClusterSpec struct {
...
+	// if empty, will use default one in namespace
+	ServiceAccountName string `json:"serviceAccountName,omitempty"`
```
Comment (functicons) [inline api/v1beta1/flinkcluster_types.go:416]: Use `*string` as the type to make it optional.

File: controllers/flinkcluster_converter.go [hunk 3]
```diff
@@ -87,6 +87,7 @@ func getDesiredJobManagerDeployment(
 	var clusterName = flinkCluster.ObjectMeta.Name
 	var clusterSpec = flinkCluster.Spec
 	var imageSpec = clusterSpec.Image
+	var serviceAccountSpec = clusterSpec.ServiceAccountName
```
Comment (functicons) [inline controllers/flinkcluster_converter.go:90]: s/serviceAccountSpec/serviceAccount

File: docs/crd.md [hunk 4]
```diff
@@ -151,6 +152,8 @@ FlinkCluster
       * **pullSecrets** (optional): Secrets for image pull.
     * **batchSchedulerName** (optional): BatchSchedulerName specifies the batch scheduler name for JobManager, TaskManager.
       If empty, no batch scheduling is enabled.
+    * **serviceAccountName** (optional): serviceAccountName specifies the service account name(which must already exist in the namespace) for JobManager, TaskManager.
```
Comment (functicons) [inline docs/crd.md:155]: the name of the service account (which must...). If empty, the default service account in the namespace will be used.

**Review by functicons — CHANGES_REQUESTED** [review]
Thanks! I left a few comments. Could you also add a test for the new field in https://github.com/GoogleCloudPlatform/flink-on-k8s-operator/blob/master/controllers/flinkcluster_converter_test.go?

File: api/v1beta1/flinkcluster_types.go (hunk 1 above)
Comment (yanghui16355) [reply api/v1beta1/flinkcluster_types.go]: fixed

File: api/v1beta1/flinkcluster_types.go (hunk 2 above)
Comment (yanghui16355) [reply api/v1beta1/flinkcluster_types.go]: fixed

File: controllers/flinkcluster_converter.go (hunk 3 above)
Comment (yanghui16355) [reply controllers/flinkcluster_converter.go]: fixed

File: docs/crd.md (hunk 4 above)
Comment (yanghui16355) [reply docs/crd.md]: fixed

**Comment by yanghui16355** [comment]
unit test for service account name added

**Comment by yanghui16355** [comment]
@functicons looks like all checks have passed, can you approve my PR then, thanks

**Review by functicons — APPROVED** [review]

**Comment by yanghui16355** [comment]
@functicons one more question, how to upgrade the flink-operator with the new version(master)? thanks

**Pull request merged**